thread.c

```diff
@@ -1355,15 +1377,20 @@ rb_thread_sleep_interruptible(void)
 }
 
 static void
-rb_thread_sleep_deadly_allow_spurious_wakeup(VALUE blocker)
+rb_thread_sleep_deadly_allow_spurious_wakeup(VALUE blocker, VALUE timeout, rb_hrtime_t *end)
```

Comment:
Are `VALUE timeout` and `rb_hrtime_t *end` the same things?

Reply:
Not quite. `timeout` is the relative duration e.g. `pop(timeout: 2)`.

I need it to pass it to the fiber scheduler if there is one.

If there is no fiber scheduler, I need the absolute timeout time, so that I can resume the sleep if we're interrupted for other reasons.